### ISSUE TITLE
Helm release fails with 'unable to parse key: assignment to entry in nil map'

### ISSUE THREAD

**Issue by idursun** [fluxcd/helm-controller#113]
We are trying to deploy linkerd with the following `HelmRelease`

```yaml
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: linkerd
  namespace: default
spec:
  chart:
    spec:
      chart: linkerd2
      sourceRef:
        kind: HelmRepository
        name: linkerd
  interval: 1m0s
  releaseName: linkerd
  targetNamespace: default
  valuesFrom:
    - kind: Secret
      name: linkerd-ca
      valuesKey: ca.crt
      targetPath: global.identityTrustAnchorsPEM
    - kind: Secret
      name: linkerd-issuer
      valuesKey: issuer.crt
      targetPath: identity.issuer.tls.crtPEM
    - kind: Secret
      name: linkerd-issuer
      valuesKey: issuer.key
      targetPath: identity.issuer.tls.keyPEM
```

We get the following error message when we run `gotk get hr -n default`

```
unable to merge value from key 'ca.crt' in Secret 'default/linkerd-ca' into target path 'global.identityTrustAnchorsPEM': unable to parse key: assignment to entry in nil map
```

We have noticed that the error is happening at this line: https://github.com/fluxcd/helm-controller/blob/6402c7d775ca6905de84e0a192ddc4e7fa42f8a5/controllers/helmrelease_controller.go#L554
as `result` is still nil.

**Comment by onatm** [comment]
Thank you for the fix @hiddeco! I couldn't find your release cycle details on the documentation. Do you have any preview releases? If you don't, I'd be glad to know when is the next release date 😃

**Comment by stefanprodan** [comment]
This fix has been released in gotk v0.1.7, update the CLI and rerun install or bootstrap to upgrade the controllers.
